rfcs/0000-execute-builds-in-standard-containers.md

```diff
@@ -0,0 +1,132 @@
+## Problem
+
+Currently, kpack executes each stage of the [CNB Lifecycle][cnb-lifecycle] inside of
+individual [init containers][init-containers]. This has two major benefits:
+
+1. Different stages of the lifecycle are isolated from one another allowing the credentials required to push to the
+   registry to be only mounted in the containers that need them, preventing a rogue [BuildPack][buildpack] from reading
+   the secrets.
+
+2. We can very easily define the order of the steps occurring as well as waiting for the previous step to complete
+   before continuing on. Additionally, failures in one init container cause the entire build to fail without additional
+   logic on our part.
+
+As kpack has grown, we have seen a need to support [service meshes][istio-service-mesh] such as [Istio][istio], which do
+not work when you have init containers that need to reach out to the network. This is
+a [well documented][istio-init-container-compatability] technical limitation related to
+Istio's [automatic sidecar injection][istio-sidecar-injection]. The limitation is due to Istio adding a sidecar
+container to the kpack build pod which runs [envoy proxy][envoy-proxy] and sets up traffic from the build containers to
+be routed through the proxy. Due to the fact that init containers run before any normal container can start, kpack
```

Comment:
Q: it makes sense to me that the limitation is linked to the fact that sidecars are regular containers, but wouldn't this limitation be on Kubernetes (i.e. the way that `initContainers` relate to regular containers)?

Reply:
Ya it is a kubernetes limitation, but the main reason that Istio is causing issues for us is because their init container runs before ours and sets the iptables rules for all containers to use the istio-proxy, but when our init container based builds start, the proxy is not started since it is a regular container. This results in our init containers not being able to access the outside internet

Reply:
Ooohh got it!